### PR TITLE
09. Test years filter

### LINKED ISSUE #16: 09. Test years filter

### PR DESCRIPTION
**Pull request by maksym-bielyshev**

### CODE REVIEW

File: tests/test_category_page.py [hunk 1]
```diff
@@ -36,3 +39,31 @@ def test_price_filter(self) -> None:
...
+    @pytest.mark.parametrize(
+        'language, year, _',
+        get_test_data_dictreader('years_filter.csv'))
+    def test_categories_dropdown_transition_without_naming(
+            self, language, year, _) -> None:
+        """Check if a specific year presents in product title.
+
+        Steps:
+            1. Open "Category" page
+            2. Choose a year
```
Comment (denyskulykov) [inline tests/test_category_page.py]: typo - `an year`

File: tests/test_category_page.py [hunk 2]
```diff
@@ -36,3 +39,31 @@ def test_price_filter(self) -> None:
...
+            3. Click on the search button
+
+        Expected result:
+            1. Year presents in all titles
```
Comment (denyskulykov) [inline tests/test_category_page.py]: Fix

File: tests/test_category_page.py [hunk 3]
```diff
@@ -36,3 +39,31 @@ def test_price_filter(self) -> None:
...
+        """
+        self.category_page.switch_proper_language(language)
+        self.category_page.year_from_dropdown.choose_dropdown_option(year)
+        self.category_page.year_to_dropdown.choose_dropdown_option(year)
+        self.category_page.click_search_link()
+        titles = self.category_page.get_titles()
+        while True:
```
Comment (denyskulykov) [inline tests/test_category_page.py]: Fix

File: tests/test_category_page.py [hunk 4]
```diff
@@ -28,11 +31,40 @@ def test_price_filter(self) -> None:
...
+        Expected result:
+            1. Selected year presents in all titles
+        """
+        self.category_page.switch_proper_language(language)
+        self.category_page.year_to_dropdown.choose_dropdown_option(year)
+        self.category_page.year_from_dropdown.choose_dropdown_option(year)
+        self.category_page.click_search_link()
+
+        assert self.category_page.if_proper_year_in_titles(year)
+        self.driver.execute_script(
```
Comment (denyskulykov) [inline tests/test_category_page.py]: Move to Base Page

File: pages/category_page.py [hunk 5]
```diff
@@ -665,5 +669,39 @@ def get_prices(self):
...
+            link = self._driver.find_element(
+                *LocatorsCategoryPage.NEXT_PAGE_LINK)
+            attribute = link.get_attribute("class")
+        return "disabled" in attribute
+
+    def if_proper_year_in_titles(self, year):
+        for title in self.get_titles():
+            return year in title
+
+    def if_price_in_proper_range(self, price_from, price_to):
```
Comment (denyskulykov) [inline pages/category_page.py]: Move this logic to test

File: pages/category_page.py [hunk 6]
```diff
@@ -665,5 +669,39 @@ def get_prices(self):
...
+            title = card.find_element(*LocatorsCategoryPage.PRODUCT_TITLE).text
+            titles.append(title)
+        return titles
+
+    def right_link(self):
+        link = self._driver.find_element(*LocatorsCategoryPage.NEXT_PAGE_LINK)
+        attribute = link.get_attribute("class")
+        return attribute
+
+    def if_disabled_navigation_link(self, navigation_link):
```
Comment (denyskulykov) [inline pages/category_page.py]: rename to "is_ ... "

File: pages/category_page.py [hunk 7]
```diff
@@ -665,5 +669,39 @@ def get_prices(self):
...
+        if navigation_link == "previous":
+            link = self._driver.find_element(
+                *LocatorsCategoryPage.PREVIOUS_PAGE_LINK)
+            attribute = link.get_attribute("class")
+        if navigation_link == "next":
+            link = self._driver.find_element(
+                *LocatorsCategoryPage.NEXT_PAGE_LINK)
+            attribute = link.get_attribute("class")
+        return "disabled" in attribute
```
Comment (denyskulykov) [inline pages/category_page.py]:
```
if navigation_link == "previous":
         locator = LocatorsCategoryPage.PREVIOUS_PAGE_LINK
elif navigation_link == "next":
         locator = LocatorsCategoryPage.NEXT_PAGE_LINK
        
            link = self._driver.find_element(
                *localtor)
            attribute = link.get_attribute("class")
            return "disabled" in attribute
```

File: tests/test_category_page.py [hunk 8]
```diff
@@ -28,11 +31,40 @@ def test_price_filter(self) -> None:
...
+            "window.scrollTo(0, document.body.scrollHeight);")
+
+        while not self.category_page.if_disabled_navigation_link("next"):
+            self.category_page.click_next_page_link()
+            assert self.category_page.if_proper_year_in_titles(year)
```
Comment (denyskulykov) [inline tests/test_category_page.py]: Add scroll

File: pages/category_page.py [hunk 9]
```diff
@@ -665,5 +669,39 @@ def get_prices(self):
...
+
+    def if_proper_year_in_titles(self, year):
+        for title in self.get_titles():
```
Comment (denyskulykov) [inline pages/category_page.py]: Check that
add debug data
self.get_titles() -> ...

File: pages/category_page.py [hunk 10]
```diff
@@ -665,5 +669,39 @@ def get_prices(self):
...
         for card in self.product_cards:
             price = card.find_element(*LocatorsCategoryPage.PRODUCT_PRICE).text
-            prices.append(int(''.join(price.split(' '))))
+            try:
+                prices.append(int(''.join(price.split(' '))))
+            except ValueError:
+                break
         return prices
+
+    def get_titles(self):
```
Comment (denyskulykov) [inline pages/category_page.py]: delete this method

File: tests/test_category_page.py [hunk 11]
```diff
@@ -28,11 +31,40 @@ def test_price_filter(self) -> None:
...
+        self.category_page.year_to_dropdown.choose_dropdown_option(year)
+        self.category_page.year_from_dropdown.choose_dropdown_option(year)
+        self.category_page.click_search_link()
+
+        assert self.category_page.if_proper_year_in_titles(year)
+        self.driver.execute_script(
+            "window.scrollTo(0, document.body.scrollHeight);")
+
+        while not self.category_page.if_disabled_navigation_link("next"):
+            self.category_page.click_next_page_link()
```
Comment (denyskulykov) [inline tests/test_category_page.py:71]: for card in self.product_cards:
            assert year in card.title

File: tests/test_category_page.py (hunk 4 above)
Comment (maksym-bielyshev) [reply tests/test_category_page.py]: Moved to "Base Test"

**Pull request merged**